### PR TITLE
alternative to #18185

### PR DESCRIPTION
**Pull request by ringabout**
close #18185

## Pros
The rlocks module is more consistent with locks module, so they can replace each other seamlessly.
## Cons
The error messages may be worse as #12231 said(link error instead of compilation error without threads options)

## Other possible solution:
make `threads:on` options default may solve this problem(ref https://github.com/nim-lang/Nim/pull/10781 and https://github.com/nim-lang/RFCs/issues/361)

### CODE REVIEW

File: lib/core/rlocks.nim [hunk 1]
```diff
@@ -11,7 +11,10 @@
 
 
 when not compileOption("threads") and not defined(nimdoc):
-  {.error: "Rlocks requires --threads:on option.".}
+  when false:
+    # make rlocks modlue consistent with locks module,
+    # so they can replace each other seamlessly.
+    {.error: "Rlocks requires --threads:on option.".}
```
Comment (timotheecour) [inline lib/core/rlocks.nim:17]: nit: pre-existing but `{.error: "rlocks requires --threads:on option.".}` (lowercase r) would make more sense

Comment (ringabout) [reply lib/core/rlocks.nim:17]: It is consistent with pre-existing marks
"ThreadPool"
"Locks"
...

**Review by timotheecour — APPROVED** [review]
LGTM for following reasons:

if we start requiring threads:on for sharedtables, it's a rabbithole, other modules would then also require `threads:on` even if they don't need to (refs https://github.com/nim-lang/Nim/pull/18185#issuecomment-855210004)

threads:on can have negative performance implications when a single thread is needed, as well cause other issues eg around gcsafety for otherwise working code.

sharedtables may be designed for threads:on but it doens't mean it should require it, and in fact `nim r --threads:off tests/stdlib/tsharedtable.nim` has been working again since https://github.com/nim-lang/Nim/pull/12331.

rlocks (like locks) should indeed work with threads:off, so that apps can support threads:on|off without having to patch a lot of code with things like `when compileOption("threads"): ...`

I'm not sure what linker error was observed (refs https://github.com/nim-lang/Nim/pull/12231#issuecomment-855647640, no details were provided), but if there's a repro we can always find a solution to avoid the link error that doesn't involve erroring for `import rlocks` for threads:off

File: lib/core/rlocks.nim (hunk 1 above)
Comment (timotheecour) [reply lib/core/rlocks.nim:17]: there is not `Rlocks` object name though... it's the whole module that (before this PR) was supposed to be disabled; anyway it's now disabled, and it's a nit

**Pull request merged**